contracts/EntityCoreFacet.sol

```diff
@@ -51,6 +52,71 @@ import "./Policy.sol";
 
 
   // IEntityCoreFacet
+  function createPolicyWithTranches(
+    //Policy Variables
+    uint256[] calldata _dates,
+    address _unit,
+    uint256 _premiumIntervalSeconds,
+    uint256[] calldata _commmissionsBP,
+    address[] calldata _stakeholders,
+    //Tranche Variable Arrays
+    // uint256[] calldata _numShares,
+    // uint256[] calldata _pricePerShareAmount,
+    // uint256[][] calldata _premiums
+    uint256[][] calldata _trancheData
+  )
+    external
+    override
+  {
+    // address[] memory stakeholders;
+    // stakeholders = new address[](6);
+    // stakeholders[0] = address(this);
+    // stakeholders[1] = msg.sender;
+    // stakeholders[2] = _stakeholders[0];
+    // stakeholders[3] = _stakeholders[1];
+    // stakeholders[4] = _stakeholders[2];
+    // stakeholders[5] = _stakeholders[3];
+
+    // require(
+    //   IAccessControl(stakeholders[2]).aclContext() == aclContext(), 
+    //   'underwriter ACL context must match'
+    // );
+    address f = this.createPolicy(
+      _dates,
+      _unit,
+      _premiumIntervalSeconds,
+      _commmissionsBP,
+      _stakeholders 
+    );
+
+    uint256[] memory premiums;
+    uint256 trancheDataLength;
+
+    IPolicy pol = IPolicy(f);
+    uint256 numTranches = _trancheData.length;
+    for (uint i=0; i<numTranches; i++) {
+      trancheDataLength = _trancheData[i].length;
+      for (uint j=2; j<trancheDataLength; ++j)
+      {
```

Comment:
Bracket should be on end of prev line, please follow code formatting in rest of file